File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> prefix;
        int sum = 0;
        int count = 0;
        prefix[0]=1;
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            if(prefix.find(sum-k)!=prefix.end()){
                count+=prefix[sum-k];
            }
            prefix[sum]++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):

        count = 0 
        size = len(nums)
        prefixSum = 0
        prefixSumMpp= defaultdict(int)
        prefixSumMpp[0]=1
        for i in range(0,size):
            prefixSum += nums[i]
            remaining = prefixSum - k
            count += prefixSumMpp[remaining]
            prefixSumMpp[prefixSum]+=1
        return count

        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        